core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -362,11 +362,13 @@ private[spark] class Worker(
             self,
             workerId,
             host,
+            webUiPort,
```

Comment:
I think that we can fix this for ephemeral ports by using `webUi.boundPort` here instead.